cpu/kinetis_common/periph/rtt.c

```diff
@@ -126,12 +126,15 @@ void rtt_set_counter(uint32_t counter)
 
 void rtt_set_alarm(uint32_t alarm, rtt_cb_t cb, void *arg)
 {
+    /* The alarm is triggered when TSR matches TAR, and TSR increments. This
+     * seem counterintuitive as most users expect the alarm to trigger
+     * immediately when the counter becomes equal to the alarm time. */
     RTC_Type *rtt = RTT_DEV;
 
     /* Disable Timer Alarm Interrupt */
     rtt->IER &= ~(RTC_IER_TAIE_MASK);
 
-    rtt->TAR = alarm;
+    rtt->TAR = alarm - 1;
```

Comment:
What happens if one passes `0` as alarm ? I doesn't make sense but maybe we should ensure `alarm` is greater than 1 ?

Reply:
It should actually still be correct, the alarm will be set to trigger when the counter is equal to 0xffffffff and increments, i.e. when the counter turns 0.